web_ui/src/pages/project-details/components/project-dataset/dataset-import-to-existing-project/keypoint-error-message.component.tsx

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2022-2025 Intel Corporation
+// LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+import { Flex, Text } from '@geti/ui';
+import { Alert } from '@geti/ui/icons';
+import { isEmpty } from 'lodash-es';
+
+import { Label } from '../../../../../core/labels/label.interface';
+import { isNonEmptyArray } from '../../../../../shared/utils';
+import {
+    getDuplicates,
+    getMissingLabels,
+    KEYPOINT_ANNOTATION_WARNING,
+    KEYPOINT_DUPLICATED_LABELS,
+    KEYPOINT_MISSING_LABELS,
+} from './utils';
+
+export interface KeypointErrorProps {
+    labelsMap: Record<string, string>;
+    labels: Label[];
+}
+
+const concatNames = (labels: Label[]) => labels.map(({ name }) => name).join(', ');
+
+export const KeypointErrorMessage = ({ labels, labelsMap }: KeypointErrorProps) => {
+    const labelsMapValues = Object.values(labelsMap);
+    const duplicatedValues = getDuplicates(labelsMapValues);
+
+    const missingLabels = getMissingLabels(labels, labelsMap);
+    const duplicatedLabels = labels.filter((label) => duplicatedValues.includes(label.id));
+
+    if (isEmpty(missingLabels) && isEmpty(duplicatedLabels)) {
+        return <></>;
+    }
+
+    return (
+        <Flex gap={'size-100'} alignContent={'center'} height={'100%'} alignItems={'center'}>
+            <Alert style={{ fill: 'var(--brand-coral-cobalt)', flex: 'none' }} />
+            <Flex direction={'column'} gap={'size-100'}>
```

Comment:
should we move the `duplicatedLabels` and `missingLabels` verification to the parent? To avoid any edge case where the Alert icon would be there but without any message? Also, np: should we make it Warning (yellow) instead of red?

Reply:
I didn’t want to move it to the parent because the `DatasetImportToExistingProjectDialogButtons ` already has a well-defined responsibility. Encapsulating the error handling here feels like a better way to maintain the separation of concerns.

As for the single icon case, I’ve added an early-return validation to prevent that scenario.

And I agree about the yellow color, red feels too serious.